### PR TITLE
Add a teal lexer

### PR DESCRIPTION
**Pull request by iscke**
This PR adds a lexer for teal (https://developer.algorand.org/docs/reference/teal/specification/), the language used for smart contracts for the algorand blockchain.

I've tried to follow the coding standards used elsewhere in this project as closely as possible, but if anything's wrong please don't hesitate to let me know. Hope this can help.

### CODE REVIEW

File: pygments/lexers/teal.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+class TealLexer(RegexLexer):
+    """
+    For the `Transaction Execution Approval Language (TEAL)
+    <https://developer.algorand.org/docs/reference/teal/specification/>`
+
+    For more information about the grammar, see:
+    https://github.com/algorand/go-algorand/blob/master/data/transactions/logic/assembler.go
+
+    .. versionadded: 2.8
```
Comment (Anteru) [inline pygments/lexers/teal.py:25]: 2.9 by now :)

**Review by Anteru — CHANGES_REQUESTED** [review]
Thanks for the contribution! Please change the tests to use the new test system using golden files, it makes it easier to maintain. (See Contributing.md for details.)

**Comment by Anteru** [comment]
Thanks! A few more things: You have to move _all_ examples into /examplefiles/teal (i.e. also tests/examplefiles/example.teal), and it looks like you're missing the golden test files (you can generate them using py.test --update-goldens)

**Comment by iscke** [comment]
oops, thank you

**Comment by Anteru** [comment]
Thanks. Just to make sure, `7777777777777777777777777777777777777777777777777774MSJUVU` is really supposed to be a single number?

**Comment by iscke** [comment]
Yeah, it's a crypto account address

**Comment by Anteru** [comment]
Merged, thanks!

**Comment by iscke** [comment]
Thank you!

**Pull request merged**